### PR TITLE
ENH remove pkg_resources; update tests

### LINKED ISSUE #360: pkg_resources is deprecated
I'm getting deprecation warnings from importing fitsio lately.  At least on 3.10 (I didn't try other versions).  

It looks like the whole pkg_resources API is deprecated now and fitsio imports it as a top-line import, even though AFAICT it's only used in testing.

The easiest fix would be to move the import into the testGZIPTileCompressedReadLosslessAstropy function, so it would only trigger if someone ran that function.  (Presumably just the CI test suite I suspect.)  But probably better to figure out what the new blessed API is for this and switch over.

To replicate:
```
$ python -W error -c "import fitsio"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/Mike/mambaforge/envs/py3.10/lib/python3.10/site-packages/fitsio/__init__.py", line 39, in <module>
    from . import test
  File "/Users/Mike/mambaforge/envs/py3.10/lib/python3.10/site-packages/fitsio/test.py", line 7, in <module>
    from pkg_resources import resource_filename
  File "/Users/Mike/mambaforge/envs/py3.10/lib/python3.10/site-packages/pkg_resources/__init__.py", line 121, in <module>
    warnings.warn("pkg_resources is deprecated as an API", DeprecationWarning)
DeprecationWarning: pkg_resources is deprecated as an API
```

### PR DESCRIPTION
**Pull request by beckermr**
Fixes #360

### CODE REVIEW

**Comment by beckermr** [comment]
@rmjarvis Care to take a look?

**Review by rmjarvis — APPROVED** [review]
Looks fine to me.

FWIW, another option I've seen that might (?) be cleaner for this is pathlib.Path.  Then the pattern is
```
from pathlib import Path
gzip_file = Path(__file__).parent / 'test_images/test_gzip_compressed_image.fits.fz'
```
I think that's fairly portable and not deprecated...  :)  Might be cleaner than the importlib way of doing it.

**Comment by beckermr** [comment]
@esheldon Any comments?

File: fitsio/test.py [hunk 1]
```diff
@@ -1296,8 +1295,19 @@ def testGZIPTileCompressedReadLosslessAstropy(self):
         """
         Test reading an image gzip compressed by astropy (fixed by cfitsio 3.49)
         """
-        gzip_file = resource_filename(__name__, 'test_images/test_gzip_compressed_image.fits.fz')
-        data = fitsio.read(gzip_file)
+        try:
+            import importlib.resources
+            ref = importlib_resources.files("fitsio") / 'test_images' / 'test_gzip_compressed_image.fits.fz'
```
Comment (esheldon) [inline fitsio/test.py:1300]: If we ever want to support windows, we should be using path joining
```
import os
path = os.path.join(
    importlib_resources.files("fitsio"),
    'test_images',
    'test_gzip_compressed_image.fits.fz',
)
```

**Review by esheldon — CHANGES_REQUESTED** [review]

File: fitsio/test.py (hunk 1 above)
Comment (beckermr) [reply fitsio/test.py:1300]: This is pathlib. The / is an overloaded python operator, not a character. This is equivalent to os.path.join

Comment (esheldon) [reply fitsio/test.py:1300]: Thanks, sorry about that. I was mentally thinking of the code Mike posted `gzip_file = Path(__file__).parent / 'test_images/test_gzip_compressed_image.fits.fz'
`

Comment (rmjarvis) [reply fitsio/test.py:1300]: That works the same way I believe.

Comment (rmjarvis) [reply fitsio/test.py:1300]: Except it should be `gzip_file = Path(__file__).parent / 'test_images' / 'test_gzip_compressed_image.fits.fz'`

**Comment by beckermr** [comment]
Any other items here @esheldon?

**Review by esheldon — APPROVED** [review]

**Pull request merged**